adafruit_thermal_printer/thermal_printer.py

```diff
@@ -287,7 +297,7 @@ def reset(self):
         # ESC + 'D' + tab stop value list ending with null to terminate.
         self.send_command("\x1BD\x04\x08\x10\x14\x18\x1C\x00")
 
-    def print(self, text, end="\n"):
+    def print(self, text: str, end: str = "\n") -> None:
```

Comment:
It looks like this function checks `end` to see if it is `None`, so it's reasonable to expect it could be.  Let's annotated `end` as `Optional`.